insights/parsers/jboss_version.py

```diff
@@ -1,46 +1,60 @@
 """
-JBoss version - File ``$JBOSS_HOME/version.txt``
-================================================
-
-This module provides plugins access to file ``$JBOSS_HOME/version.txt``
-
-Typical content of file ``$JBOSS_HOME/version.txt`` is::
-
-    Red Hat JBoss Enterprise Application Platform - Version 6.4.3.GA
-
-This module parses the file content and stores data in the dict ``self.parsed``.
-The version info can also be got via ``obj.major`` and ``obj.minor``, etc.
-
-Examples:
-    >>> jboss_version.file_path
-    '/home/test/jboss/jboss-eap-6.4/version.txt'
-    >>> jboss_version.raw
-    'Red Hat JBoss Enterprise Application Platform - Version 6.4.3.GA'
-    >>> jboss_version.major
-    6
-    >>> jboss_version.minor
-    4
-    >>> jboss_version.release
-    3
-    >>> jboss_version.version
-    '6.4.3'
-    >>> jboss_version.code_name
-    'GA'
+JBoss version
+=============
+Provide information about the versions of all running Jboss on a system.
 """
-from .. import Parser, parser
-from ..specs import Specs
+
+import json
+
+from insights import Parser, parser
+from insights.specs import Specs
+from insights.core.context import Context
 
 
 @parser(Specs.jboss_version)
 class JbossVersion(Parser):
-    """Parses the content of file ``$JBOSS_HOME/version.txt``."""
+    """
```

Comment:
@chenlizhong since the spec `jboss_version` is no longer being collected this parser should be deprecated.  What do you think about pulling the parse_content out into a function and then just calling that function from both the deprecated parser and the new parser?

Reply:
This jboss_version is to parse <jboss_home>/version.txt. Insights-client does not collect it. But It is in JDR archive. I'd prefer to keep it.  The new parser (JbossRuntimeVersions) is just to parse multiple version.txt files.  One version.txt is corresponding to one instance of JbossVersion. So it is nature to use the JbossVersion class directly and do not need the seperate function of parse_content.

Reply:
I had not checked the JDR archive specs.  In that case I agree we should keep it.

Reply:
Updated